### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by meztez**
Just kidding,

Even thought it exposes some uglier part of plumber, I added an example on how to programmatically add routes with params and parsers. Hopefully this will help some people.

### CODE REVIEW

File: R/parse-body.R [hunk 1]
```diff
@@ -149,12 +149,12 @@ parser_picker <- function(content_type, first_byte, filename = NULL, parsers = N
 #'
 #' @details
 #' When `parser` is evaluated, it should return a parser function.
-#' Parser matching is done first by `content-type` header matching on `fixed` then by using a
-#' regular expressions on `regex`. Note that plumber strip the header from `; charset*` to
-#' perform matching.
+#' Parser matching is done first by `content-type` header matching with `fixed` then by using
+#' regular expressions with `regex`. Note that plumber strip `; charset*` from `content-type` header before
+#' performing match.
```
Comment (schloerke) [inline R/parse-body.R:154]:
```suggestion
#' regular expressions with `regex`. Note that plumber strips `; charset*` from `content-type` header before matching.
```

File: R/parse-body.R [hunk 2]
```diff
@@ -149,12 +149,12 @@ parser_picker <- function(content_type, first_byte, filename = NULL, parsers = N
...
 #'
-#' There is a special case when no `content-type` header is
-#' provided that will use a [parser_json()] when it detects a `json` string.
+#' It will try to use [parser_json()] if available when no `content-type` header is found and
+#' request body starts with `{` or `[`.
```
Comment (schloerke) [inline R/parse-body.R:157]:
```suggestion
#' Plumber will try to use [parser_json()] (if available) when no `content-type` header is found and
#' the request body starts with `{` or `[`.
```

File: R/parse-body.R [hunk 3]
```diff
@@ -347,20 +347,19 @@ make_parser <- function(aliases) {
...
+#' functions when enabling them on router. This will allow for
 #' non-default behavior.
 #'
-#' Parsers are optional. When unspecified, only the [parser_json()],
-#' [parser_octet()], [parser_form()] and [parser_text()] are available.
-#' You can use `@parser parser` tag to activate parsers per endpoint.
-#' Multiple parsers can be activated for the same endpoint using multiple `@parser parser` tags.
+#' Parsers are optional. When unspecified, only default endpoint parsers are enabled.
+#' You can use `@parser parser` tag to enable parser on endpoint.
+#' Multiple parsers can be enabled on the same endpoint using multiple `@parser parser` tags.
```
Comment (schloerke) [inline R/parse-body.R:357]:
```suggestion
#' You can use `@parser NAME` tag to enable parser on endpoint.
#' Multiple parsers can be enabled on the same endpoint using multiple `@parser NAME` tags.
```

File: R/parse-body.R [hunk 4]
```diff
@@ -347,20 +347,19 @@ make_parser <- function(aliases) {
...
 #'
 #' User should be aware that `rds` parsing should only be done from a
 #' trusted source. Do not accept `rds` files blindly.
 #'
-#' See [registered_parsers()] for a list of registered parsers.
+#' See [registered_parsers()] for a list of registered parsers aliases.
```
Comment (schloerke) [inline R/parse-body.R:362]:
```suggestion
#' See [registered_parsers()] for a list of registered parsers names.
```

File: R/plumb.R [hunk 5]
```diff
@@ -4,7 +4,7 @@
 #' taking an incoming request, submitting it through the appropriate filters and
 #' eventually to a corresponding endpoint, if one is found.
 #'
-#' See \url{https://www.rplumber.io/articles/programmatic-usage.html} for additional
+#' See [here](../articles/programmatic-usage.html) for additional
```
Comment (schloerke) [inline R/plumb.R:7]:
```suggestion
#' See the [Programmatic Usage](../articles/programmatic-usage.html) article for additional
```

File: R/plumber.R [hunk 6]
```diff
@@ -34,7 +34,7 @@ defaultPlumberFilters <- list(
 #' taking an incoming request, submitting it through the appropriate filters and
 #' eventually to a corresponding endpoint, if one is found.
 #'
-#' See \url{https://www.rplumber.io/articles/programmatic-usage.html} for additional
+#' See [here](../articles/programmatic-usage.html) for additional
```
Comment (schloerke) [inline R/plumber.R:37]:
```suggestion
#' See the [Programmatic Usage](../articles/programmatic-usage.html) article for additional
```

File: R/pr_set.R [hunk 7]
```diff
@@ -21,6 +21,7 @@ pr_set_serializer <- function(pr, serializer) {
 #' default parsers for any endpoint that does not define their own parsers.
 #'
 #' Note: The default set of parsers will be completely replaced if any value is supplied. Be sure to include all of your parsers that you would like to include.
+#' Use `registered_parsers()` to get a list of available parsers aliases.
```
Comment (schloerke) [inline R/pr_set.R:24]:
```suggestion
#' Use `registered_parsers()` to get a list of available parser names.
```

File: R/serializer.R [hunk 8]
```diff
@@ -103,7 +105,7 @@ as_attachment <- function(value, filename = NULL) {
 #'
 #' Serializers are used in Plumber to transform the R object produced by a
 #' filter/endpoint into an HTTP response that can be returned to the client. See
-#' [here](https://book.rplumber.io/articles/rendering-output.html#serializers-1) for
+#' [here](../articles/rendering-output.html#serializers-1) for
```
Comment (schloerke) [inline R/serializer.R:108]: CRAN will believe the relative link is an error. 😒 

```suggestion
#' [here](https://www.rplumber.io/articles/rendering-output.html#serializers-1) for
```

File: vignettes/annotations.Rmd [hunk 9]
```diff
@@ -107,13 +107,32 @@ function(name, age) {
...
+         handler = f, 
+         serializer = serializer_text(),
+         parsers = "none",
+         responses = list("200" = list(description = "A sentence"))) %>%
+  pr_post(path = "/upload_file",
+          handler = g,
+          serializer = serializer_rds(),
+          parsers = c("multi", "rds"),
+          params = list("f" = list(type = "file", desc = "A file", required = FALSE, isArray = FALSE)),
+          comments = "Upload an rds file and return the object")
```
Comment (schloerke) [inline vignettes/annotations.Rmd:132]:
```suggestion
text_handler <- function(name, age) { sprintf("%s is %i years old", name, age) }
file_handler <- function(file) { as_attachment(file[[1]], names(file)[1]) }
pr() %>%
  pr_get(path = "/dyn/<name:str>/<age:[int]>/route", 
         handler = text_handler, 
         serializer = serializer_text(),
         parsers = "none",
         responses = list("200" = list(description = "A sentence"))) %>%
  pr_post(path = "/upload_file",
          handler = file_handler,
          serializer = serializer_rds(),
          parsers = c("multi", "rds"),
          params = list("file" = list(type = "file", desc = "A file", required = FALSE, isArray = FALSE)),
          comments = "Upload an rds file and return the object")
```

File: vignettes/annotations.Rmd [hunk 10]
```diff
@@ -107,13 +107,32 @@ function(name, age) {
...
 #* Upload an rds file and return the object
 function(f) {
   as_attachment(f[[1]], names(f)[1])
 }
 ```
 
+##### Equivalent programmatic usage
+```{r, eval = FALSE, echo = TRUE}
+f <- function(name, age) { sprintf("%s is %i years old", name, age) }
+g <- function(f) { as_attachment(f[[1]], names(f)[1]) }
```
Comment (schloerke) [inline vignettes/annotations.Rmd:120]: These programmatic usage sections are AWESOME! Whenever they can be added (or makes sense to add), I'm all for it!

File: vignettes/rendering-output.Rmd [hunk 11]
```diff
@@ -44,6 +44,12 @@ JSON is not appropriate for every situation, however. If you want your API to re
 
 By default, Plumber serializes objects into JSON via the `jsonlite` R package. However, there are a variety of other serializers that are built in to the package.
 
+You can also pass arguments to certain serializers to modify their behavior like in the example below.
+```r
+#* @serializer json list(na="string")
+```
+See [here](../reference/serializers.html) for possible arguments.
```
Comment (schloerke) [inline vignettes/rendering-output.Rmd:51]:
```suggestion
See the [Serialization](../reference/serializers.html) article for details.
```

File: vignettes/routing-and-input.Rmd [hunk 12]
```diff
@@ -306,7 +306,9 @@ e <- r$endpoints[[1]][[1]]
 code_chunk(json_serialize(e$exec(req=list(bodyRaw = charToRaw("id=123&name=Jennifer"), body=list(id = 123, name = "Jennifer"), args = list(id = 123, name = "Jennifer")), res = NULL)), "json")
 ```
 
-Alternatively, `curl --data '{"id":123, "name": "Jennifer"}' "http://localhost:8000/user"` (formatting the body as JSON) will have the same effect.
+Alternatively, `echo {"id":123, "name": "Jennifer"} > call.json & curl --data @call.json "http://localhost:8000/user" -H "content-type: application/json"` (formatting the body as JSON) will have the same effect.
+
+Plumber
```
Comment (schloerke) [inline vignettes/routing-and-input.Rmd:311]:
```suggestion
```

**Review by schloerke — CHANGES_REQUESTED** [review]
Minor tweaks on the docs.  Thank you so much!

File: R/serializer.R (hunk 8 above)
Comment (meztez) [reply R/serializer.R:108]: Oh well

File: vignettes/rendering-output.Rmd (hunk 11 above)
Comment (meztez) [reply vignettes/rendering-output.Rmd:51]: Is CRAN going to complain here too?

Comment (schloerke) [reply vignettes/rendering-output.Rmd]: Dang. Yes. Missed it. Updating the comment above

File: R/plumb.R [hunk 13]
```diff
@@ -4,7 +4,7 @@
 #' taking an incoming request, submitting it through the appropriate filters and
 #' eventually to a corresponding endpoint, if one is found.
 #'
-#' See \url{https://www.rplumber.io/articles/programmatic-usage.html} for additional
+#' See the [Programmatic Usage](../articles/programmatic-usage.html) article for additional
```
Comment (schloerke) [inline R/plumb.R:7]:
```suggestion
#' See the [Programmatic Usage](https://www.rplumber.io/articles/programmatic-usage.html) article for additional
```

File: vignettes/rendering-output.Rmd [hunk 14]
```diff
@@ -44,6 +44,12 @@ JSON is not appropriate for every situation, however. If you want your API to re
 
 By default, Plumber serializes objects into JSON via the `jsonlite` R package. However, there are a variety of other serializers that are built in to the package.
 
+You can also pass arguments to certain serializers to modify their behavior like in the example below.
+```r
+#* @serializer json list(na="string")
+```
+See the [Serialization](../reference/serializers.html) article for details.
```
Comment (schloerke) [inline vignettes/rendering-output.Rmd:51]:
```suggestion
See the [Serialization](https://www.rplumber.io/reference/serializers.html) article for details.
```

File: R/plumber.R [hunk 15]
```diff
@@ -34,7 +34,7 @@ defaultPlumberFilters <- list(
 #' taking an incoming request, submitting it through the appropriate filters and
 #' eventually to a corresponding endpoint, if one is found.
 #'
-#' See \url{https://www.rplumber.io/articles/programmatic-usage.html} for additional
+#' See the [Programmatic Usage](../articles/programmatic-usage.html) article for additional
```
Comment (schloerke) [inline R/plumber.R:37]:
```suggestion
#' See the [Programmatic Usage](https://www.rplumber.io/articles/programmatic-usage.html) article for additional
```

**Review by schloerke — APPROVED** [review]
Using fully defined links

**Pull request merged**